costmap_converter/src/costmap_to_polygons.cpp

```diff
@@ -215,10 +215,12 @@ void CostmapToPolygonsDBSMCCH::updateCostmap2D()
       for (auto& n : neighbor_lookup_)
         n.clear();
 
+      auto size_x = costmap_->getSizeInCellsX();
+      auto size_y = costmap_->getSizeInCellsY();
       // get indices of obstacle cells
-      for(std::size_t i = 0; i < costmap_->getSizeInCellsX(); i++)
+      for(std::size_t i = 0; i < size_x; i++)
       {
-        for(std::size_t j = 0; j < costmap_->getSizeInCellsY(); j++)
+        for(std::size_t j = 0; j < size_y; j++)
```

Comment:
https://stackoverflow.com/questions/1242185/loop-condition-evaluation

I confirmed with Valgrind profiling that it is more performant